core/src/com/po/fuck/model/collections/CollidableCollection.java

```diff
@@ -13,6 +14,11 @@ public class CollidableCollection extends SimpleCollection<Collidable> {
     }
 
     public List<Collidable> collides(Polygon other) {
+        if(other == null) return Collections.emptyList();
+        return objects.stream().filter(x -> !x.isInvisible() && x.collide(other)).collect(Collectors.toList());
+    }
+
+    public List<Collidable> collidesWithInvisible(Polygon other){
```

Comment:
`collidesWithInvisible` this name should be changed, because it checks collision not with only invisible.